src/lib.rs

```diff
@@ -178,3 +180,49 @@ impl Write for Uart {
 
 // SAFETY: `Uart` just contains a pointer to device memory, which can be accessed from any context.
 unsafe impl Send for Uart {}
+
+impl ErrorType for Uart {
+    type Error = Infallible;
+}
+
+impl Write for Uart {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, Self::Error> {
```

Comment:
Just curious, any reason we are writing only one byte at a time? 
Also, should we implement write_all as well?

Reply:
Writing a single byte is sufficient to satisfy the specification of `Write::write`, and avoids blocking any more than necessary. If there's a more efficient way to write multiple bytes than polling like this in a loop then we can do that.

`write_all` is a provided method, i.e. the trait has a default implementation in terms of `write`. We don't need to implement it unless there's a more efficient way to do so than the default.

Reply:
Got it, thanks!